### ISSUE TITLE
EnforceNotMet: the provided recursive_sequence_lengths info is invalid 

### ISSUE THREAD

**Issue by cqu-ls** [PaddlePaddle/Paddle#18882]
使用时序数据的时候发现，当样本特征维度是1的时候，可以正常运行。但是特征维度大于1的时候会报错（使用reader_creator 产生随机训练样本），请问应该怎样设置reader_creator（）中输入数据的维度，以及fluid.layers.data中的shape应该怎样设置？假如我要输入到LSTM中的数据：时间步是4，batch是4，样本特征是5，应该如何设置？
import paddle.fluid as fluid
import numpy as np
#import random
import paddle

BATCH_SIZE = 4
MAX_TIME = 2

def reader_creator():
    def reader():
        while True:
            yield np.random.randint(100,size=**2**),np.random.randint(1,size=1)
    return reader
  

emb_dim = 256
vocab_size = 10000
hidden_dim = 512

data = fluid.layers.data(name='data', shape=[-1,**2**],
               dtype='float32', lod_level=1)
#emb = fluid.layers.embedding(input=data, size=[vocab_size, emb_dim], is_sparse=True)

forward_proj = fluid.layers.fc(input=data, size=hidden_dim * 4,
                               bias_attr=False)
forward, _ = fluid.layers.dynamic_lstm(
    input=forward_proj, size=hidden_dim * 4, use_peepholes=False)

out_put = fluid.layers.sequence_last_step(input=forward)
out_put = fluid.layers.fc(input=out_put, size=1,bias_attr=False)

label = fluid.layers.data(name="label", shape=[1], dtype="float32",lod_level=1)
cost = fluid.layers.square_error_cost(input=out_put, label=label)
avg_cost = fluid.layers.mean(cost)

#accuracy = fluid.layers.accuracy(input=out_put, label=label)
fluid.optimizer.Adagrad(learning_rate=0.002).minimize(avg_cost)
 
main_program = fluid.default_main_program() # 获取默认/全局主函数
startup_program = fluid.default_startup_program() # 获取默认/全局启动程序  

use_cuda = False
place = fluid.CUDAPlace(0) if use_cuda else fluid.CPUPlace() # 指明executor的执行场所
exe = fluid.Executor(place)
exe.run(startup_program)

train_reader = paddle.batch(paddle.reader.shuffle(reader_creator(),buf_size= 10),batch_size=BATCH_SIZE)
feeder = fluid.DataFeeder(place=place, feed_list=[data,label])

for data_train in train_reader():
    avg_loss_value= exe.run(main_program,feed=feeder.feed(data_train),fetch_list=[avg_cost.name])
    print(avg_loss_value)

**Comment by cqu-ls** [comment]
已经解决

**Comment by cqu-ls** [comment]
已解决
